lms/djangoapps/learner_dashboard/tests/test_views.py

```diff
@@ -93,10 +94,15 @@ def test_program_discussion_disabled(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.data, expected_data)
 
-    def test_api_returns_discussions_iframe(self):
+    @ddt.data(True, False)
+    def test_api_returns_discussions_iframe(self, staff):
         """
-        Test if API returns iframe in case ProgramDiscussionsConfiguration model contains proper data
+        Test if API returns iframe in case ProgramDiscussionsConfiguration model contains proper data for staff
+        and non staff. In case of non staff user must be be enrolled in program.
         """
+        if staff:
+            self.user = UserFactory(is_staff=True)
+            self.client.login(username=self.user.username, password=self.password)
```

Comment:
`self.user = UserFactory(is_staff=staff)`

Reply:
This will fail because it will create a new user with no enrolment in case of non staff user